BootloaderCorePkg/Library/SmbiosInitLib/SmbiosInitLib.c

```diff
@@ -379,6 +410,48 @@ AddSmbiosType (
 }
 
 
+/**
+  This function is called to initialize the SmbiosStringsPtr.
+**/
+VOID
+EFIAPI
+InitSmbiosStringPtr (
+  VOID
+  )
+{
+
+  EFI_STATUS            Status;
+  UINT16                Length;
+  VOID                 *SmbiosStringsPtr;
+
+  //
+  // Allocate the memmory for SMBIOS String Ptr,
+  //
+
+  SmbiosStringsPtr = AllocateZeroPool ((PcdGet16(PcdSmbiosStringsCnt) * sizeof (SMBIOS_TYPE_STRINGS)) );
+  if (SmbiosStringsPtr == NULL) {
+    DEBUG ((DEBUG_INFO, "SmbiosStringsPtr Memory allocation failed"));
+    ASSERT_EFI_ERROR(EFI_OUT_OF_RESOURCES);
+  }
+
+  //
+  // Copy contents from Default String Array
+  //
+
+  Length = sizeof (mDefaultSmbiosStrings);
+  if(Length <= (PcdGet16(PcdSmbiosStringsCnt) * sizeof (SMBIOS_TYPE_STRINGS)))
+    CopyMem (SmbiosStringsPtr, mDefaultSmbiosStrings, Length);
+  else {

```

Comment:
```
if () {
} else {
}
```

Reply:
Do we need to use braces for single line in if-else state ? pls confirm

Reply:
Ok will do it. checked in other code. Braces are used in all if - else statement

Reply:
Unlike Linux coding convention, EDKII must use braces regardless of the number of lines.

FYI, here is EDKII coding standard.
https://edk2-docs.gitbook.io/edk-ii-c-coding-standards-specification/v/release%2F2.20/3_quick_reference#3-2-formatting